### PR TITLE
fix: semver checking

### PR DESCRIPTION
**Pull request by fzipi**
In some cases we need to use a "short version".

👉 This somehow got into a different PR, I'm raising it here so it is visible.

### CODE REVIEW

**Comment by fzipi** [comment]
wth. pushed one additional commit, and it is not showing up 🤔

File: regex/definitions.go [hunk 1]
```diff
@@ -81,6 +81,12 @@ var DefinitionReferenceRegex = regexp.MustCompile(`{{([a-zA-Z0-9-_]+)}}`)
 // The version declared on the file is captured in group 1.
 var CRSVersionRegex = regexp.MustCompile(`^(# OWASP (ModSecurity Core Rule Set|CRS) ver\.)(.+)$`)
 
+// ShortCRSVersionRegex matches the version contained on every rules file.
+// The version declared on the file is captured in group 1.
+// This regex is used to match the version in the setup file.
+// As numbers can grow to something like 10.12.123, the range is set to 3-8 to capture bigger numbers.
```
Comment (theseion) [inline regex/definitions.go:87]: There's no range in the regex...?

File: regex/definitions.go [hunk 2]
```diff
@@ -81,6 +81,12 @@ var DefinitionReferenceRegex = regexp.MustCompile(`{{([a-zA-Z0-9-_]+)}}`)
 // The version declared on the file is captured in group 1.
 var CRSVersionRegex = regexp.MustCompile(`^(# OWASP (ModSecurity Core Rule Set|CRS) ver\.)(.+)$`)
 
+// ShortCRSVersionRegex matches the version contained on every rules file.
+// The version declared on the file is captured in group 1.
+// This regex is used to match the version in the setup file.
```
Comment (theseion) [inline regex/definitions.go:86]:
```suggestion
```

File: regex/definitions.go [hunk 3]
```diff
@@ -81,6 +81,12 @@ var DefinitionReferenceRegex = regexp.MustCompile(`{{([a-zA-Z0-9-_]+)}}`)
...
+// As numbers can grow to something like 10.12.123, the range is set to 3-8 to capture bigger numbers.
+var ShortCRSVersionRegex = regexp.MustCompile(`(setvar:tx.crs_setup_version=)(\d+)`)
+
 // CRSCopyrightYearRegex matches the version and year range of the copyright text in setup,
 // setup example, and rule files.
 // The matched end year of the copyright year range will be captured in group 1.
```
Comment (theseion) [inline regex/definitions.go:92]: In all the regular expressions below, the year / version is actually in capture group 2, not 1. Could you update the comments?

**Review by theseion — CHANGES_REQUESTED** [review]

File: regex/definitions.go (hunk 1 above)
Comment (fzipi) [reply regex/definitions.go:87]: Yeah, I moved to `\d+`. Removing.

File: regex/definitions.go [hunk 4]
```diff
@@ -71,34 +71,32 @@ var RuleIdFileNameRegex = regexp.MustCompile(`^(\d{6})(?:-chain(\d+))?(?:\.ra)?$
 var RuleIdTestFileNameRegex = regexp.MustCompile(`^(\d{6})(?:\.ya?ml)?$`)
 
 // TestTitleRegex matches any test_title line in test YAML files (test_title: "<title>").
-// Everything up to the value of the test title is captured in group 1.
+// Everything up to the value of the test title is captured in group 1, test title in group 2.
```
Comment (theseion) [inline regex/definitions.go:74]: the title isn't captured.

**Review by theseion — DISMISSED** [review]

File: regex/definitions.go (hunk 4 above)
Comment (fzipi) [reply regex/definitions.go:74]: Now it is! 🥳

**Review by theseion — APPROVED** [review]

**Pull request merged**